### ISSUE TITLE
Potato search result issue

### ISSUE THREAD

**Issue by MrFreeRobot** [Jackett/Jackett#4816]
Hello,

I'm having issues with couchpotato and jackett.
In couchpotato log, I have the following log error:
`http://jackett:9117/api/v2.0/indexers/yggtorrent/results/potato/?imdbid=tt0115658&user=&passkey=xxx&user=&passkey=xxx`
I doubled checked passkey, it's all good. The URL is the one provided in Jackett UI. `jackett`points to a local DNS.
Using the url in the browser pops a `401 error`.
Messing a bit with URLs I have the following results:

`http://jackett:9117/api/v2.0/indexers/yggtorrent/results/?user=&passkey=xxxx&imdbid=tt0983193` works ok, even though results don't match
`http://jackett:9117/api/v2.0/indexers/yggtorrent/results/potato/?user=&passkey=xxxx&imdbid=tt0983193` (notes that the only difference is the **potato** befoore parameters) gives the following result `yggtorrent does not support the requested query. Please check the capabilities (t=caps) and make sure the search mode and categories are supported.`

**Jackett version**:  0.10.880.0
**Mono version** (if not using Windows): Using windows  

Any suggestions ? Anything I'm doing wrong ?
Thank you

**Comment by garfield69** [comment]
I believe that ygg does not support imdbid searches, so i don't think you will ever get it to with with couchpotato.

**Comment by MrFreeRobot** [comment]
Isn't Jackett supposed to use OMDB to apped movie name based on IMDB id ? 

I also noticed that the 401 comes from the fact that the ùser`and `passkey` are used twice in the URL. I think that's a pure couchpotato issue...

**Comment by garfield69** [comment]
oh, is that what the omdb api key is for, I didn't know that.
ok, so presuming that you have loaded the omdb api key in jackett, is it only ygg you are having trouble with?
have you tried any of the other public jackett indexers, are they all working?

**Comment by MrFreeRobot** [comment]
Yes, I tried with 1337x which is public, same result.
`http://jackett:9117/api/v2.0/indexers/yggtorrent/results/potato/?search=avengers&user=&passkey=xxx`
or
`http://jackett:9117/api/v2.0/indexers/1337x/results/potato/?search=avengers&user=&passkey=xxx`
both work.
The search with imdbid fails:
`http://jackett117/api/v2.0/indexers/1337x/results/potato/?imdbid=tt4154756&user=&passkey=xxx`

**Comment by garfield69** [comment]
@kaso17  or @flightlevel (or anyone else in the know) can you clarify for me what the omdb api is supposed to be used for, and aside from that, whether or not couchpotato calls via jackett only use imdbid searches?
if the latter is true, then does this mean that only indexers that support direct imdb searches will work with couchpotato, or is that what the omdb api is suppose to help with?
Your feedback appreciated, thanks.

**Comment by garfield69** [comment]
OK, I finally read the readme, and found https://github.com/Jackett/Jackett#configuring-omdb so I now know that the omdb is used for.
I still don't know how to debug this issue so I'm going to pass it on to more capable devs.

**Comment by garfield69** [comment]
@MrFreeRobot  can you confirm that your omdb key is valid by using this test
 `http://www.omdbapi.com/?i=tt3896198&apikey=yourkey`
you should get something like this back
```json
{"Title":"Guardians of the Galaxy Vol. 2","Year":"2017","Rated":"PG-13","Released":"05 May 2017","Runtime":"136 min","Genre":"Action, Adventure, Comedy, Sci-Fi","Director":"James Gunn","Writer":"James Gunn, Dan Abnett (based on the Marvel comics by), Andy Lanning (based on the Marvel comics by), Steve Englehart (Star-Lord created by), Steve Gan (Star-Lord created by), Jim Starlin (Gamora and Drax created by), Stan Lee (Groot created by), Larry Lieber (Groot created by), Jack Kirby (Groot created by), Bill Mantlo (Rocket Raccoon created by), Keith Giffen (Rocket Raccoon created by), Steve Gerber (Howard the Duck created by), Val Mayerik (Howard the Duck created by)","Actors":"Chris Pratt, Zoe Saldana, Dave Bautista, Vin Diesel","Plot":"The Guardians struggle to keep together as a team while dealing with their personal family issues, notably Star-Lord's encounter with his father the ambitious celestial being Ego.","Language":"English","Country":"USA","Awards":"Nominated for 1 Oscar. Another 12 wins & 42 nominations.","Poster":"https://m.media-amazon.com/images/M/MV5BMTg2MzI1MTg3OF5BMl5BanBnXkFtZTgwNTU3NDA2MTI@._V1_SX300.jpg","Ratings":[{"Source":"Internet Movie Database","Value":"7.7/10"},{"Source":"Rotten Tomatoes","Value":"84%"},{"Source":"Metacritic","Value":"67/100"}],"Metascore":"67","imdbRating":"7.7","imdbVotes":"449,175","imdbID":"tt3896198","Type":"movie","DVD":"22 Aug 2017","BoxOffice":"$389,804,217","Production":"Walt Disney Pictures","Website":"https://marvel.com/guardians","Response":"True"}
```
just to eliminate the possibility of an invalid omdb key being the issue ;-)

**Comment by garfield69** [comment]
I've done some testing and have the following to report:
I got a key for the omdbapi and added it to the jackett dashboard settings.

then I install couchpotato 3.0.1 and set up the searches page with torrentpotato entries for yggtorrent, zooqle, yts and rarbg (all of which are public jackett indexers, the first two do not support imdbid and the latter two do).

as far as I can tell, the torrentpotato interface on couchpotato will only search successfully on jackett indexers that support imdbid searches.

on jackett the public indexers yts and rarbg support this type of search and these yielded results for couchpotato.

trying to use couchpotato's torrentcouch with jackett's yggtorrent or zooqle (both of which do not have imdbid support) yields errors.

there are other private indexers that support imdbid but I did not bother testing with these.

I don't know if adding a omdbkey to jackett is supposed to enable general searching for couchpotato on all jackett indexers or not, but judging from my tests, it seems currently that only those indexers that support imdbid searching work with couchpotato.

**Comment by garfield69** [comment]
@MrFreeRobot  did some more research. 
turns out that cp 3.0.1 which I downloaded from https://couchpota.to/ is a release build in sep 2015 and is 550 commits behind the source code at https://github.com/CouchPotato/CouchPotatoServer/

After uninstalling that version, I followed the run-from-source instructions at https://github.com/CouchPotato/CouchPotatoServer/#couchpotato

now cp can search any of the jackett indexers. tested with yggtorrent and zooqle.
